src/analysis/special_functions/gamma.lean

```diff
@@ -0,0 +1,587 @@
+end
+
+/- This one too -- a more general version allowing exp(-bx) for any b > 0 -/
+lemma tendsto_exp_mul_div_rpow_at_top (s : ℝ) (b : ℝ) (hb : 0 < b):
+  tendsto (λ x : ℝ, exp (b * x) / x ^ s ) at_top at_top :=
+begin
+  have t1 := tendsto_exp_div_rpow_at_top (s/b),
+  have t2 := tendsto_rpow_at_top hb,
+  have t := tendsto.comp t2 t1,
+
+  let f1 := (λ (x : ℝ), (exp x / x ^ (s / b)) ^ b),
+  let f2 := (λ x : ℝ, exp (b * x) / x ^ s ),
+  have ff1 : ∀ x:ℝ, f1 x = (exp x / x ^ (s / b)) ^ b,
+  { by simp only [eq_self_iff_true, forall_const] },
+  have ff2 : ∀ x:ℝ, f2 x = exp (b * x) / x ^ s,
+  {by simp only [eq_self_iff_true, forall_const] },
+
+  have Ioieq: eq_on f1 f2 (Ioi 0),
+  { intros x hx,
+    rw [set.Ioi, mem_set_of_eq] at hx,
+    rw [ff1, ff2],
+    rw div_rpow,
+    -- clean up the subgoals introduced by div_rpow
+    show 0 ≤ exp x,
+    { apply le_of_lt(exp_pos x) },
+    show 0 ≤ x ^ (s / b),
+    { apply le_of_lt,
+      apply rpow_pos_of_pos hx },
+
+    rw [←(exp_mul x b), mul_comm x b],
+    rw div_eq_div_iff,
+    show x^s ≠ 0,
+    { symmetry, apply ne_of_lt,
+      apply rpow_pos_of_pos,
+      linarith },
+    show (x ^ (s / b)) ^ b ≠ 0,
+    { symmetry, apply ne_of_lt,
+      apply rpow_pos_of_pos,
+      apply rpow_pos_of_pos,
+      linarith },
+    rw mul_eq_mul_left_iff,
+    left,
+    rw ←rpow_mul,
+    show 0 ≤ x, linarith,
+    rw div_mul_cancel, exact hb.ne' },
+  have : Ici (1:ℝ) ⊆ Ioi (0:ℝ),
+  { rw [set.Ioi, set.Ici],
+    intro x,
+    rw mem_set_of_eq,rw mem_set_of_eq,
+    intro hx,linarith },
+  have Ioi_at_top: Ioi (0:ℝ) ∈ at_top := Ioi_mem_at_top (0:ℝ),
+  have ev_eq: eventually_eq at_top f1 f2 := eventually_eq_of_mem Ioi_at_top Ioieq,
+
+  exact tendsto.congr' ev_eq t
+end
+
+lemma tendsto_exp_mul_div_rpow_at_top' (s : ℝ) (b : ℝ) (hb : 0 < b):
+  tendsto (λ x : ℝ, x^s * exp (-b * x)) at_top (𝓝 $ (0:ℝ)) :=
+begin
+  have l := tendsto_exp_mul_div_rpow_at_top s b hb,
+  have:  (λ x : ℝ, x^s * exp (-b * x)) =  (λ x : ℝ, exp (b * x) / x^s)⁻¹,
+  { ext,
+    simp only [neg_mul, pi.inv_apply],
+    rw [inv_div,div_eq_mul_inv],
+    rw mul_eq_mul_left_iff,
+    left,
+    apply exp_neg },
+  rw this,
+  exact tendsto.inv_tendsto_at_top l
+end
+
+/- Now we have the bits we need -/
+lemma asymp_F (s : ℝ) :
+  asymptotics.is_O s.F (λ x : ℝ, exp(-(1/2) * x)) filter.at_top :=
+begin
+  apply asymptotics.is_o.is_O,
+  apply asymptotics.is_o_of_tendsto,
+  { intros x hx,
+    exfalso,
+    apply ( exp_pos(-(1/2) * x)).ne',
+    exact hx },
+  simp only [F],
+
+  have : ∀ (x: ℝ), (x > 0) → (exp (-x) * x ^ s / exp (-(1 / 2) * x) = exp (-(1/2) * x) * x ^ s),
+  { intros x h,
+    rw mul_comm,
+    rw mul_comm (exp (-(1/2) * x)) (x ^ s),
+    rw div_eq_of_eq_mul,
+    exact (exp_pos (-(1/2) * x)).ne',
+    have: exp(-x) = exp(-(1/2)*x) * exp (-(1 / 2) * x),
+    { rw ←real.exp_add,
+      simp only [exp_eq_exp],
+      ring },
+    rw this,
+    ring },
```

Comment:
```suggestion
    rw [this, mul_assoc], },
```
`ring` is a bit overkill here, using high power tactics is often slower (when checking the proof) than basic ones, so if they are only doing a couple of steps its best just to do the steps to by hand.